crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2603,6 +2604,59 @@ impl<'db> TypeInferenceBuilder<'db> {
                     op,
                     Type::IntLiteral(i64::from(b)),
                 ),
+
+            (Type::StringLiteral(salsa_s1), Type::StringLiteral(salsa_s2)) => {
+                // For anything non-ASCII, don't even try to infer precise literal types.
+                // It would be complicated, it's an unimportant edge case,
+                // and Rust and Python probably do subtly different things
+                // in their string methods for non-ASCII characters.
+                let s1 = salsa_s1.value(self.db);
+                if !s1.is_ascii() {
+                    return Some(builtins_symbol_ty(self.db, "bool").to_instance(self.db));
+                }
+                let s2 = salsa_s2.value(self.db);
+                if !s2.is_ascii() {
+                    return Some(builtins_symbol_ty(self.db, "bool").to_instance(self.db));
+                }
```

Comment:
heh, now that you've accepted my suggestion, maybe we should add some tests with some emojis or something 😆